storage/redis.go

```diff
@@ -64,6 +74,100 @@ func createRedisDatabase(config RedisConfig) (*redisDatabase, error) {
 	if !isRedisURL(config.Address) {
 		config.Address = "redis://" + config.Address
 	}
+
+	// Redis Sentinel support: if sentinelMasterName is present in the connection URL, crete a Sentinel client.
+	sentinelOptions, err := parseRedisSentinelURL(config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to configure Redis Sentinel client: %w", err)
+	}
+	if sentinelOptions != nil {
+		return &redisDatabase{
+			sentinelOptions: sentinelOptions,
+			databaseName:    config.Database,
+		}, nil
+	}
+
+	// Regular Redis client
+	opts, err := parseRedisConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &redisDatabase{
+		options:      opts,
+		databaseName: config.Database,
+	}, nil
+}
+
+// parseRedisSentinelURL parses connectionString as URL to see if Redis Sentinel support must be enabled.
+// If so, it returns true, the parsed URL without Sentinel options (because otherwise redis.ParseURL() would fail later on),
+// If the connectionString doesn't specify Redis Sentinel options, it returns false.
+func parseRedisSentinelURL(config RedisConfig) (*redis.FailoverOptions, error) {
+	//uriString := "redis://host1:1234,host2:4321?sentinelMasterName=bla"
+	// Errors return by url.Parse() are ignored, because they only happen in edge, extremely edge situations.
+	// We just return from the function, and the error occurs and gets captured again when using redis.ParseURL().
+	sentinelURI, _ := url.Parse(config.Address)
+	if sentinelURI == nil {
+		return nil, nil
+	}
+	masterName := sentinelURI.Query().Get(sentinelMasterNameParam)
+	if len(masterName) == 0 {
+		// Sentinel not enabled
+		return nil, nil
```

Comment:
This quietly ignores the other two Sentinel params. A warning would be nice?

A `func sentinelParams(*url.URL) (hosts []string, master, user, pass string)` would help here. This also eliminates the needs for the constant declarations on line 37–39.

Reply:
I agree; https://github.com/nuts-foundation/nuts-node/pull/1504